recipes/cpython/all/conanfile.py

```diff
@@ -0,0 +1,544 @@
+        solution_path = os.path.join(self._source_subfolder, "PCBuild", "pcbuild.sln")
+        projects = set(m.group(1) for m in re.finditer("\"([^\"]+)\\.vcxproj\"", open(solution_path).read()))
+        discarded = self._msvc_discarded_projects
+
+        def project_build(name):
+            if os.path.basename(name) in discarded:
+                return False
+            if "test" in name:
+                return False
+            return True
+
+        projects = set(p for p in projects if project_build(p))
+        return projects
+
+    @property
+    def _msvc_discarded_projects(self):
+        discarded = {"python_uwp", "pythonw_uwp"}
+        if not self.options.with_bz2:
+            discarded.add("bz2")
+        if not self.options.with_sqlite3:
+            discarded.add("_sqlite3")
+        if not self.options.with_tkinter:
+            discarded.add("_tkinter")
+        if self._is_py2:
+            # Python 2 Visual Studio projects NOT to build
+            discarded = discarded.union({"bdist_wininst", "libeay", "ssleay", "sqlite3", "tcl", "tk", "tix"})
+            if not self.options.with_bsddb:
+                discarded.add("_bsddb")
+        elif self._is_py3:
+            discarded = discarded.union({"bdist_wininst", "liblzma", "openssl", "sqlite3", "xxlimited"})
+            if not self.options.with_lzma:
+                discarded.add("_lzma")
+        return discarded
+
+    @property
+    def _msvc_archs(self):
+        archs = {
+            "x86": "Win32",
+            "x86_64": "x64",
+        }
+        if tools.Version(self.version) >= "3.8":
+            archs.update({
+                "armv7": "ARM",
+                "armv8_32": "ARM",
+                "armv8": "ARM64",
+            })
+        return archs
+
+    def _msvc_build(self):
+        msbuild = MSBuild(self)
+        msbuild_properties = {
+            "IncludeExternals": "true",
+        }
+        projects = self._solution_projects
+        self.output.info("Building {} Visual Studio projects: {}".format(len(projects), projects))
+
+        upgraded = False
+        with tools.no_op():
+            for project_i, project in enumerate(projects):
+                self.output.info("[{}/{}] Building project '{}'...".format(project_i, len(projects), project))
+                project_file = os.path.join(self._source_subfolder, "PCBuild", project + ".vcxproj")
+                msbuild.build(project_file, upgrade_project=not upgraded, build_type="Debug" if self.settings.build_type == "Debug" else "Release",
+                              platforms=self._msvc_archs, properties=msbuild_properties)
+                upgraded = True
+
+    def build(self):
+        if tools.Version(self.version) < "3.8.0":
+            if tools.Version(self.deps_cpp_info["mpdecimal"].version) >= "2.5.0":
+                raise ConanInvalidConfiguration("cpython versions lesser then 3.9.0 require a mpdecimal lesser then 2.5.0")
+        else:
```

Comment:
This should now be an `elif` checking that the cpython version is >= 3.9.0.

Reply:
It can, but I'm going to keep it as it is.
It's imho easier to reason about this code, and the runtime overhead is non-existent.

Reply:
I think the test is incorrect now, as cpython 3.8 allows both mpdecimal 2.4 and 2.5.
It's only cpython 3.9 that requires mpdecimal 2.5

Reply:
I applied your suggestion. It was correct after all.